---
title: "Coexpression Networks for DEG PPI Network"
author: "Claire Green"
date: "16/10/2017"
output: html_document
---

#Generating the Coexpression Networks

To generate the coexpression networks I took the rows from each normalised expression matrix that correspond to the PPI network nodes. I then used Spearman's Rank to generate correlation values for all gene-gene relationships. I chose Spearman's rank because the microarray values are relative measures and the RNA-seq values are absoulute measures so aren't necessarily direclty comparable. As Spearman correlates by rank rather than by absoulte value this seemed like a better option.

*This takes a very long time to calculate. Anywhere between half an hour to 2 hours for each of my small datasets. If this is done in a larger dataset then it's worth looking into making this more efficient*

```{r, eval=FALSE}
#### DEG PPI Correlation ####

#Read in network nodes
DEG_PPI <- readLines("~/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/DEG_PPI_Genes.txt")

#Extract PPI network genes from each dataset
C9 <- read.csv("~/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/Microarray_AllGenesExpression/C9uniquegene_samples.csv", row.names = 1)
C9 <- C9[,4:11]
C9 <- subset(C9, rownames(C9) %in% DEG_PPI)

VCP <- read.csv("~/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/Microarray_AllGenesExpression/VCPuniquegene_samples.csv", row.names = 1)
VCP <- VCP[,4:10]
VCP <- subset(VCP, rownames(VCP) %in% DEG_PPI)

FTLD <- read.csv("~/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/Microarray_AllGenesExpression/FTLDuniquegene_samples.csv", row.names = 1)
FTLD <- FTLD[,9:24]
FTLD <- subset(FTLD, rownames(FTLD) %in% DEG_PPI)

sALS <- read.csv("~/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/Microarray_AllGenesExpression/sALSuniquegene_samples.csv", row.names = 1)
sALS <- sALS[,4:10]
sALS <- subset(sALS, rownames(sALS) %in% DEG_PPI)

PET <- read.csv("~/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/TDP-43_DEseq2/PET_results_keepfiltering.csv")
rownames(PET) <- PET$hgnc_symbol
PET <- PET[,19:35]
PET <- subset(PET, rownames(PET) %in% DEG_PPI)

RAV <- read.csv("~/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/TDP-43_DEseq2/RAV_results_keepfiltering.csv")
rownames(RAV) <- RAV$hgnc_symbol
RAV <- RAV[,18:30]
RAV <- subset(RAV, rownames(RAV) %in% DEG_PPI)

#Find the gene names that all datasets have in common
DEG_com <- Reduce(intersect, list(rownames(C9), rownames(VCP), rownames(FTLD),
                                  rownames(sALS), rownames(PET), rownames(RAV)))

#Subset each dataset with these common names so they are all the same size
C9 <- subset(C9, rownames(C9) %in% DEG_com)
VCP <- subset(VCP, rownames(VCP) %in% DEG_com)
FTLD <- subset(FTLD, rownames(FTLD) %in% DEG_com)
sALS <- subset(sALS, rownames(sALS) %in% DEG_com)
PET <- subset(PET, rownames(PET) %in% DEG_com)
RAV <- subset(RAV, rownames(RAV) %in% DEG_com)

#### Cor.test Method ####

library(tictoc)
library(gdata)

##For loop for generating regression values and p values
CorExprMat <- t(VCP)

test <- CorExprMat

reg <- matrix(0, ncol(test), ncol(test))
p.value <- matrix(0, ncol(test), ncol(test))

tic()
for (i in 1:ncol(test)){
  for (j in 1:ncol(test)){
    reg[i,j] <- cor.test(test[,i], test[,j], method = "spearman")$estimate
  }}

rownames(reg) <- colnames(reg) <- colnames(test)
toc()

tic()
for (i in 1:ncol(test)){
  for (j in 1:ncol(test)){
    p.value[i,j] <- cor.test(test[,i], test[,j], method = "spearman")$p.value
  }}

rownames(p.value) <- colnames(p.value) <- colnames(test)
toc()


##Only take upper triangle without diagonal (all comparisons are currently doubled)
ptri <- p.value
ptri[lower.tri(ptri, diag = TRUE)] <- NA

#Turn into vector
p.vec <- unmatrix(ptri)
#Remove NA values
p.vec <- na.omit(p.vec)
#Multiple hypothesis testing correction
p.adj <- p.adjust(p.vec, method = "fdr", n = length(p.vec))

#Create results table
reg.mat <- unmatrix(reg)
reg.mat <- as.data.frame(reg.mat)
p.adj <- as.data.frame(p.adj)
p.mat <- as.data.frame(p.vec)

pvals <- merge(p.adj, p.mat, by.x = "row.names", by.y = "row.names")
rownames(pvals)<- pvals$Row.names
pvals[,1] <- NULL
results <- merge(pvals, reg.mat, by.x = "row.names", by.y = "row.names")
rownames(results)<- results$Row.names
results[,1] <- NULL
results <- results[order(results$p.vec),]

setwd("~/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/")
write.csv(reg.mat, "VCP_Rho_PPI_coexpression.csv")
```

From this I created a file for each dataset with the gene pair, the p value, the adjusted p value and the Rho value. I then filtered the datasets by only taking the relationships where the Rho value was either above 0.5 or below -0.5.

```{R, eval = F}
#### filter correlations ####

#Split names to two columns
C9_cor <- read.csv("C9_PPI_coexpression.csv")
C9_cor$Gene1 <- as.character(lapply(strsplit(as.character(C9_cor$X), "\\:"), "[", 2))
C9_cor$Gene2 <- as.character(lapply(strsplit(as.character(C9_cor$X), "\\:"), "[", 1))
C9_cor <- C9_cor[,c(5,6,2,3,4)]

sALS_cor <- read.csv("sALS_PPI_coexpression.csv")
sALS_cor$Gene1 <- as.character(lapply(strsplit(as.character(sALS_cor$X), "\\:"), "[", 2))
sALS_cor$Gene2 <- as.character(lapply(strsplit(as.character(sALS_cor$X), "\\:"), "[", 1))
sALS_cor <- sALS_cor[,c(5,6,2,3,4)]

FTLD_cor <- read.csv("FTLD_PPI_coexpression.csv")
FTLD_cor$Gene1 <- as.character(lapply(strsplit(as.character(FTLD_cor$X), "\\:"), "[", 2))
FTLD_cor$Gene2 <- as.character(lapply(strsplit(as.character(FTLD_cor$X), "\\:"), "[", 1))
FTLD_cor <- FTLD_cor[,c(5,6,2,3,4)]

VCP_cor <- read.csv("VCP_Rho_PPI_coexpression.csv")
VCP_cor$Gene1 <- as.character(lapply(strsplit(as.character(VCP_cor$X), "\\:"), "[", 2))
VCP_cor$Gene2 <- as.character(lapply(strsplit(as.character(VCP_cor$X), "\\:"), "[", 1))
VCP_cor <- VCP_cor[,c(5,6,2,3,4)]

PET_cor <- read.csv("PET_PPI_coexpression.csv")
PET_cor$Gene1 <- as.character(lapply(strsplit(as.character(PET_cor$X), "\\:"), "[", 2))
PET_cor$Gene2 <- as.character(lapply(strsplit(as.character(PET_cor$X), "\\:"), "[", 1))
PET_cor <- PET_cor[,c(5,6,2,3,4)]

RAV_cor <- read.csv("RAV_PPI_coexpression.csv")
RAV_cor$Gene1 <- as.character(lapply(strsplit(as.character(RAV_cor$X), "\\:"), "[", 2))
RAV_cor$Gene2 <- as.character(lapply(strsplit(as.character(RAV_cor$X), "\\:"), "[", 1))
RAV_cor <- RAV_cor[,c(5,6,2,3,4)]


### Filter by r value
C9_cor.5 <- C9_cor[C9_cor$reg.mat > 0.5 | C9_cor$reg.mat < -0.5,]
sALS_cor.5 <- sALS_cor[sALS_cor$reg.mat > 0.5 | sALS_cor$reg.mat < -0.5,]
FTLD_cor.5 <- FTLD_cor[FTLD_cor$reg.mat > 0.5 | FTLD_cor$reg.mat < -0.5,]
VCP_cor.5 <- VCP_cor[VCP_cor$reg.mat > 0.5 | VCP_cor$reg.mat < -0.5,]
PET_cor.5 <- PET_cor[PET_cor$reg.mat > 0.5 | PET_cor$reg.mat < -0.5,]
RAV_cor.5 <- RAV_cor[RAV_cor$reg.mat > 0.5 | RAV_cor$reg.mat < -0.5,]

setwd("~/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/")
write.csv(C9_cor.5, "C9_cor.5_cytoscape.csv", row.names = F, quote = F)
write.csv(sALS_cor.5, "sALS_cor.5_cytoscape.csv", row.names = F, quote = F)
write.csv(FTLD_cor.5, "FTLD_cor.5_cytoscape.csv", row.names = F, quote = F)
write.csv(VCP_cor.5, "VCP_cor.5_cytoscape.csv", row.names = F, quote = F)
write.csv(PET_cor.5, "PET_cor.5_cytoscape.csv", row.names = F, quote = F)
write.csv(RAV_cor.5, "RAV_cor.5_cytoscape.csv", row.names = F, quote = F)
```

From this I generated network files that can be uploaded into cytoscape. As Wenbin pointed out you can do the network merging in R, however I chose to do it in cytoscape as I knew I would be using those networks to merge in other ways and it seemed quicker. The downside is that cytoscape really struggles with large networks so crashes can happen. 

#THE MISTAKE
*I must write here that at first I made a mistake in my code where I indexed the wrong columns for the VCP patients. This meant I had to redo the VCP correlations all over again but the results are largely similar.*

# Coexpression Networks
| Dataset            | Platform   | Number of Edges after filtering |
|--------------------|------------|---------------------------------|
| C9orf72 ALS        | Microarray | 749485                          |
| sALS               | Microarray | 1738947                         |
| sFTLD + GRN FTLD   | Microarray | 875269                          |
| VCP                | Microarray | 1924014                         |
| C9orf72 ALS + sALS | RNA-Seq    | 907927                          |
| C9orf72 ALS        | RNA-Seq    | 591875                          |

Here is a table showing the sizes of the networks after filtering. 

I managed to merge the networks in cytoscape and this produced a collection of 365 nodes with 240 common connections. However, what I soon realised was that cytoscape classes a common edge as any edge conserved between datasets regardless of edge value. This mean that edges which are positively coexpressed in one dataset and negatively in another are included. So I had to go through and remove any edges which weren't completely unanimous in the direction of coexpression. This left me with 159 nodes with 171 connections:

```{r, out.width = "900px", out.height = "700px", echo=F}
knitr::include_graphics("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/Group1Network.pdf")
```

Obviously I was particularly interested in the central module of 104 genes and 131 connections. I conducted some investigatory analysis which I have summarised in the powerpoint presentation PPIcoexpressionResults. Essentially what it says is that there are a number of genes which are known to be involved in processes linked to neurodegeneration, including TBK1 which is an ALS/FTLD gene. 

Functional enrichment suggests immune related processes and protein degradation, both inextricably linked to neurodegeneration.

```{r, out.width = "900px", out.height = "700px", echo=F}
knitr::include_graphics("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/Screen Shot 2017-10-24 at 12.11.09 (2).png")
```

I then decided to overlap the module with the PPI network to see if there were any direct interactions as well as coexpression. Unfortunately none of the coexpression modules' edges corresponded to any protein-protein interaction edges, however when you overlay the coexpression network onto the PPI network you can see that it sits quite centrally in the PPI space:

```{r, out.width = "900px", out.height = "700px", echo=F}
knitr::include_graphics("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/PPIandCor.pdf")
```

This suggests that although the coexpressed genes are not directly interacting with one another, they are part of a smaller, more central sub-section of the PPI network that perhaps we can identify. 


#Comparison with Controls
To see if there are any differences in the connectivity of these 104 genes when you look at them in controls, I conducted the same analysis but seeding the correlation calculation with the 104 genes rather than the whole PPI network. 

```{r, eval=F}
#### Repeat for controls ####

#Read in network nodes
Group1Genes <- readLines("~/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/Group1.1Genes.txt")

#Extract PPI network genes from each dataset
C9 <- read.csv("~/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/Microarray_AllGenesExpression/C9uniquegene_samples.csv", row.names = 1)
C9 <- C9[,1:3]
C9 <- subset(C9, rownames(C9) %in% Group1Genes)

VCP <- read.csv("~/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/Microarray_AllGenesExpression/VCPuniquegene_samples.csv", row.names = 1)
VCP <- VCP[,1:3]
VCP <- subset(VCP, rownames(VCP) %in% Group1Genes)

FTLD <- read.csv("~/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/Microarray_AllGenesExpression/FTLDuniquegene_samples.csv", row.names = 1)
FTLD <- FTLD[,1:8]
FTLD <- subset(FTLD, rownames(FTLD) %in% Group1Genes)

sALS <- read.csv("~/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/Microarray_AllGenesExpression/sALSuniquegene_samples.csv", row.names = 1)
sALS <- sALS[,1:3]
sALS <- subset(sALS, rownames(sALS) %in% Group1Genes)

PET <- read.csv("~/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/TDP-43_DEseq2/PET_results_keepfiltering.csv")
rownames(PET) <- PET$hgnc_symbol
PET <- PET[,10:18]
PET <- subset(PET, rownames(PET) %in% Group1Genes)

RAV <- read.csv("~/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/TDP-43_DEseq2/RAV_results_keepfiltering.csv")
rownames(RAV) <- RAV$hgnc_symbol
RAV <- RAV[,10:17]
RAV <- subset(RAV, rownames(RAV) %in% Group1Genes)


#### Cor.test Method ####

library(tictoc)
library(gdata)

##For loop for generating regression values and p values
CorExprMat <- t(RAV)

test <- CorExprMat

reg <- matrix(0, ncol(test), ncol(test))
p.value <- matrix(0, ncol(test), ncol(test))

tic()
for (i in 1:ncol(test)){
  for (j in 1:ncol(test)){
    reg[i,j] <- cor.test(test[,i], test[,j], method = "spearman")$estimate
  }}

rownames(reg) <- colnames(reg) <- colnames(test)
toc()

tic()
for (i in 1:ncol(test)){
  for (j in 1:ncol(test)){
    p.value[i,j] <- cor.test(test[,i], test[,j], method = "spearman")$p.value
  }}

rownames(p.value) <- colnames(p.value) <- colnames(test)
toc()


##Only take upper triangle without diagonal (all comparisons are currently doubled)
ptri <- p.value
ptri[lower.tri(ptri, diag = TRUE)] <- NA

#Turn into vector
p.vec <- unmatrix(ptri)
#Remove NA values
p.vec <- na.omit(p.vec)
#Multiple hypothesis testing correction
p.adj <- p.adjust(p.vec, method = "fdr", n = length(p.vec))

#Create results table
reg.mat <- unmatrix(reg)
reg.mat <- as.data.frame(reg.mat)
p.adj <- as.data.frame(p.adj)
p.mat <- as.data.frame(p.vec)

pvals <- merge(p.adj, p.mat, by.x = "row.names", by.y = "row.names")
rownames(pvals)<- pvals$Row.names
pvals[,1] <- NULL
results <- merge(pvals, reg.mat, by.x = "row.names", by.y = "row.names")
rownames(results)<- results$Row.names
results[,1] <- NULL
results <- results[order(results$p.vec),]

setwd("~/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/ControlGroup1.1/")
write.csv(results, "RAVcon_PPI_coexpression.csv")


#### Split Names ####

C9_cor <- read.csv("C9con_PPI_coexpression.csv")
C9_cor$Gene1 <- as.character(lapply(strsplit(as.character(C9_cor$X), "\\:"), "[", 2))
C9_cor$Gene2 <- as.character(lapply(strsplit(as.character(C9_cor$X), "\\:"), "[", 1))
C9_cor <- C9_cor[,c(5,6,2,3,4)]

sALS_cor <- read.csv("sALScon_PPI_coexpression.csv")
sALS_cor$Gene1 <- as.character(lapply(strsplit(as.character(sALS_cor$X), "\\:"), "[", 2))
sALS_cor$Gene2 <- as.character(lapply(strsplit(as.character(sALS_cor$X), "\\:"), "[", 1))
sALS_cor <- sALS_cor[,c(5,6,2,3,4)]

FTLD_cor <- read.csv("FTLDcon_PPI_coexpression.csv")
FTLD_cor$Gene1 <- as.character(lapply(strsplit(as.character(FTLD_cor$X), "\\:"), "[", 2))
FTLD_cor$Gene2 <- as.character(lapply(strsplit(as.character(FTLD_cor$X), "\\:"), "[", 1))
FTLD_cor <- FTLD_cor[,c(5,6,2,3,4)]

VCP_cor <- read.csv("VCPcon_PPI_coexpression.csv")
VCP_cor$Gene1 <- as.character(lapply(strsplit(as.character(VCP_cor$X), "\\:"), "[", 2))
VCP_cor$Gene2 <- as.character(lapply(strsplit(as.character(VCP_cor$X), "\\:"), "[", 1))
VCP_cor <- VCP_cor[,c(5,6,2,3,4)]

PET_cor <- read.csv("PETcon_PPI_coexpression.csv")
PET_cor$Gene1 <- as.character(lapply(strsplit(as.character(PET_cor$X), "\\:"), "[", 2))
PET_cor$Gene2 <- as.character(lapply(strsplit(as.character(PET_cor$X), "\\:"), "[", 1))
PET_cor <- PET_cor[,c(5,6,2,3,4)]

RAV_cor <- read.csv("RAVcon_PPI_coexpression.csv")
RAV_cor$Gene1 <- as.character(lapply(strsplit(as.character(RAV_cor$X), "\\:"), "[", 2))
RAV_cor$Gene2 <- as.character(lapply(strsplit(as.character(RAV_cor$X), "\\:"), "[", 1))
RAV_cor <- RAV_cor[,c(5,6,2,3,4)]

### Filter by r value
C9_cor.5 <- C9_cor[C9_cor$reg.mat > 0.5 | C9_cor$reg.mat < -0.5,]
sALS_cor.5 <- sALS_cor[sALS_cor$reg.mat > 0.5 | sALS_cor$reg.mat < -0.5,]
FTLD_cor.5 <- FTLD_cor[FTLD_cor$reg.mat > 0.5 | FTLD_cor$reg.mat < -0.5,]
VCP_cor.5 <- VCP_cor[VCP_cor$reg.mat > 0.5 | VCP_cor$reg.mat < -0.5,]
PET_cor.5 <- PET_cor[PET_cor$reg.mat > 0.5 | PET_cor$reg.mat < -0.5,]
RAV_cor.5 <- RAV_cor[RAV_cor$reg.mat > 0.5 | RAV_cor$reg.mat < -0.5,]

setwd("~/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/ControlGroup1.1/")
write.csv(C9_cor.5, "C9_cor.5_cytoscape.csv", row.names = F, quote = F)
write.csv(sALS_cor.5, "sALS_cor.5_cytoscape.csv", row.names = F, quote = F)
write.csv(FTLD_cor.5, "FTLD_cor.5_cytoscape.csv", row.names = F, quote = F)
write.csv(VCP_cor.5, "VCP_cor.5_cytoscape.csv", row.names = F, quote = F)
write.csv(PET_cor.5, "PET_cor.5_cytoscape.csv", row.names = F, quote = F)
write.csv(RAV_cor.5, "RAV_cor.5_cytoscape.csv", row.names = F, quote = F)

#Save common module as edge file and split gene names into two columns
Group1 <- read.csv("MergedNetworkdefaultedge.csv")

C9merge <- merge(Group1, C9_cor.5, by.x = c("Gene1", "Gene2"), by.y = c("Gene1", "Gene2"))
sALSmerge <- merge(Group1, sALS_cor.5, by.x = c("Gene1", "Gene2"), by.y = c("Gene1", "Gene2"))
FTLDmerge <- merge(Group1, FTLD_cor.5, by.x = c("Gene1", "Gene2"), by.y = c("Gene1", "Gene2"))
VCPmerge <- merge(Group1, VCP_cor.5, by.x = c("Gene1", "Gene2"), by.y = c("Gene1", "Gene2"))
PETmerge <- merge(Group1, PET_cor.5, by.x = c("Gene1", "Gene2"), by.y = c("Gene1", "Gene2"))
RAVmerge <- merge(Group1, RAV_cor.5, by.x = c("Gene1", "Gene2"), by.y = c("Gene1", "Gene2"))

Group1Expr <- data.frame(row.names = C9merge$name,
                         C9 = C9merge$reg.mat.y,
                         sALS = sALSmerge$reg.mat.y,
                         FTLD = FTLDmerge$reg.mat.y,
                         VCP = VCPmerge$reg.mat.y,
                         PET = PETmerge$reg.mat.y,
                         RAV = RAVmerge$reg.mat.y)

Group1up <- subset(Group1Expr, (Group1Expr$C9 > 0.5 &
                                  Group1Expr$sALS > 0.5 &
                                  Group1Expr$FTLD > 0.5 &
                                  Group1Expr$VCP > 0.5 &
                                  Group1Expr$PET > 0.5 &
                                  Group1Expr$RAV > 0.5))

Group1down <- subset(Group1Expr, (Group1Expr$C9 < -0.5 &
                                    Group1Expr$sALS < -0.5 &
                                    Group1Expr$FTLD < -0.5 &
                                    Group1Expr$VCP < -0.5 &
                                    Group1Expr$PET < -0.5 &
                                    Group1Expr$RAV < -0.5))

Group1samedir <- rbind(Group1up, Group1down)
Group1samedir$corMean <- rowMeans(Group1samedir, na.rm = FALSE, dims = 1)
write.csv(Group1samedir, "G1_samedir_mean.csv", quote = F)
```

From this analysis, I discovered that there are only 4 interactions conserved between this set of 104 genes.
```{r, out.width = "900px", out.height = "700px", echo=F}
knitr::include_graphics("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/ControlGroup1.1/MergedNetwork.pdf")
```

What is even more interesting is that these connections *don't exist in the patient network*. So not only are the patient edges gained from the "healthy" network, but the edges in the "healthy" network are lost in the patients.

#Overlap with DEGs

I wanted to see if the module had any DEGs in it as it would be positive to see genes in there whose expression we know has changed. I discovered 7 DEGs were in the network:

| DEGs Overlap |
|--------------|
| VDAC1        |
| WASF1        |
| PPP3CA       |
| OAT          |
| HABP4        |
| CLCN6        |
| ARMC1        |

```{r, out.width = "900px", out.height = "700px", echo=F}
  knitr::include_graphics("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/G1.1_DEGs.pdf")
```

There are also 3 ALSOD genes - TBK1, ELP3 and GARS. 

#Interaction with known TDP pathology-causing genes

My next question was to ask - how many steps does it take to get from any diease mutation to one of the nodes of my network. As a quick first analysis I looked for direct binding relationships between the disease genes and the module genes. (PPI_mutandgroup1.1.R)

```{r, eval=F}
library(biomaRt)

setwd("~/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/")

iref14 <- read.table("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/iref14_Human_UP_noDup_table_nodash.txt", header = T)
braingenes <- read.csv("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Zhang_BrainCelltype_Markers_braingenes.csv", header = T)
genelist <- readLines("~/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/Group1.1Genes.txt")

mart <- useMart("ENSEMBL_MART_ENSEMBL",dataset="hsapiens_gene_ensembl", host="www.ensembl.org")
attributes <- listAttributes(mart)
mart_back <- getBM(attributes =c("hgnc_symbol", "uniprotswissprot"), filters="hgnc_symbol", values=genelist,  mart=mart)

genelist_Uniprot <- subset(mart_back, !(mart_back$uniprotswissprot == ""))
swiss <- subset(genelist_Uniprot, genelist_Uniprot$hgnc_symbol %in% genelist)
write.csv(swiss, "PPI_mutgenes/martback.csv", row.names = F)

###### IDENTIFY MISSING GENES AND FIND UNIPROT CODES FOR THEM. NB SOME GENES MAY NOT BE PROTEIN CODING#####

disgenes <- readLines("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/TDP-43genes.txt")

setwd("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/PPI_mutgenes")
mart_table <- read.csv("martback.csv", header = T)

# for (i in 1:length(disgenes)){
#   mutgene <- disgenes[i]
#   mutgene_back <- getBM(attributes =c("hgnc_symbol", "uniprotswissprot"), filters="hgnc_symbol", values=mutgene,  mart=mart)
#   martplusgene <- rbind(mart_table, mutgene_back[1,])
#   uniprot_gene <- mart_table$uniprotswissprot
#   PPI2 <- subset(iref14, iref14$V1 %in% uniprot_gene & iref14$V2 %in% uniprot_gene)
#   mutgeneprot <- mutgene_back$uniprotswissprot
#   interactions <- subset(PPI, PPI$V1 %in% mutgeneprot | PPI$V2 %in% mutgeneprot)
#   write.csv(interactions, file = paste(mutgene, "_PPI.csv", sep = ""), row.names = F, quote = F)
# }

mutgene <- disgenes
mutgene_back <- getBM(attributes =c("hgnc_symbol", "uniprotswissprot"), filters="hgnc_symbol", values=mutgene,  mart=mart)
martplusgene <- rbind(mart_table, mutgene_back)
genelist_Uniprot <- subset(martplusgene, !(martplusgene$uniprotswissprot == ""))
genelist_Uniprot <- subset(genelist_Uniprot,!(duplicated(genelist_Uniprot$hgnc_symbol)))
uniprot_gene <- genelist_Uniprot$uniprotswissprot
PPI_All <- subset(iref14, iref14$V1 %in% uniprot_gene & iref14$V2 %in% uniprot_gene)
write.csv(PPI_All, "ALL_PPI.csv", row.names = F, quote = F)


### Group1.1 only
setwd("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/PPI_mutgenes")
mart_table <- read.csv("martback.csv", header = T)
group1.1 <- mart_table$uniprotswissprot
PPI <- subset(iref14, iref14$V1 %in% group1.1 & iref14$V2 %in% group1.1)
write.csv(PPI, "Group1.1_PPI.csv", row.names = F, quote = F)
```

I then visualised the results in cytoscape. Yellow nodes are canonical disease genes, blue nodes are the module genes, and red nodes are genes that appear in both:
```{r, out.width = "900px", out.height = "700px", echo=F}
knitr::include_graphics("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/PPI_mutgenes/ALL_PPI.pdf")
```


#CORRECTION AFTER BUG ##

There was a bug in the code that meant that some of the genes that came out with common positive or negative correlations actually weren't. I have corrected this using the following code. 

Instead of:

```{r, eval=F}
C9merge <- merge(Group1, C9_cor.5, by.x = c("Gene1", "Gene2"), by.y = c("Gene1", "Gene2"))
sALSmerge <- merge(Group1, sALS_cor.5, by.x = c("Gene1", "Gene2"), by.y = c("Gene1", "Gene2"))
FTLDmerge <- merge(Group1, FTLD_cor.5, by.x = c("Gene1", "Gene2"), by.y = c("Gene1", "Gene2"))
VCPmerge <- merge(Group1, VCP_cor.5, by.x = c("Gene1", "Gene2"), by.y = c("Gene1", "Gene2"))
PETmerge <- merge(Group1, PET_cor.5, by.x = c("Gene1", "Gene2"), by.y = c("Gene1", "Gene2"))
RAVmerge <- merge(Group1, RAV_cor.5, by.x = c("Gene1", "Gene2"), by.y = c("Gene1", "Gene2"))
```

I used:

```{r, eval=F}
#Merge into one table. First data is read in, but because the genes have been separated into two columns
#we want them back in one column so that they can be cross referenced correctly. It just so happens that to match
#the cytoscape format, we have to put the gene in the second column first and first column second.

setwd("~/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/")
C9_cor.5 <- read.csv("C9_cor.5_cytoscape.csv")
C9_cor.5$combo <- paste(C9_cor.5$Gene2,":",C9_cor.5$Gene1, sep = "")

sALS_cor.5 <-read.csv("sALS_cor.5_cytoscape.csv")
sALS_cor.5$combo <- paste(sALS_cor.5$Gene2,":",sALS_cor.5$Gene1, sep = "")

FTLD_cor.5 <-read.csv("FTLD_cor.5_cytoscape.csv")
FTLD_cor.5$combo <- paste(FTLD_cor.5$Gene2,":",FTLD_cor.5$Gene1, sep = "")

VCP_cor.5 <-read.csv("VCP_cor.5_cytoscape.csv")
VCP_cor.5$combo <- paste(VCP_cor.5$Gene2,":",VCP_cor.5$Gene1, sep = "")

PET_cor.5 <-read.csv("PET_cor.5_cytoscape.csv")
PET_cor.5$combo <- paste(PET_cor.5$Gene2,":",PET_cor.5$Gene1, sep = "")

RAV_cor.5 <-read.csv("RAV_cor.5_cytoscape.csv")
RAV_cor.5$combo <- paste(RAV_cor.5$Gene2,":",RAV_cor.5$Gene1, sep = "")



Group1 <- read.csv("MergedNetworkdefaultedgeNEW.csv")
Group2 <- read.csv("SubnetworkEdgeInfo.csv")

C9merge <- merge(Group2, C9_cor.5, by.x = "X", by.y = "combo")
sALSmerge <- merge(Group2, sALS_cor.5, by.x = "X", by.y = "combo")
FTLDmerge <- merge(Group2, FTLD_cor.5, by.x = "X", by.y = "combo")
VCPmerge <- merge(Group2, VCP_cor.5, by.x = "X", by.y = "combo")
PETmerge <- merge(Group2, PET_cor.5, by.x = "X", by.y = "combo")
RAVmerge <- merge(Group2, RAV_cor.5, by.x = "X", by.y = "combo")

#Original module
# Group1 <- read.csv("MergedNetworkdefaultedgeNEW.csv")
#Recalculated module (CORRECT VERSION)
Group2 <- read.csv("SubnetworkEdgeInfo.csv")

C9merge <- merge(Group2, C9_cor.5, by.x = "X", by.y = "combo")
sALSmerge <- merge(Group2, sALS_cor.5, by.x = "X", by.y = "combo")
FTLDmerge <- merge(Group2, FTLD_cor.5, by.x = "X", by.y = "combo")
VCPmerge <- merge(Group2, VCP_cor.5, by.x = "X", by.y = "combo")
PETmerge <- merge(Group2, PET_cor.5, by.x = "X", by.y = "combo")
RAVmerge <- merge(Group2, RAV_cor.5, by.x = "X", by.y = "combo")

Group1Expr <- data.frame(row.names = C9merge$name,
                         C9 = C9merge$reg.mat.y,
                         sALS = sALSmerge$reg.mat.y,
                         FTLD = FTLDmerge$reg.mat.y,
                         VCP = VCPmerge$reg.mat.y,
                         PET = PETmerge$reg.mat.y,
                         RAV = RAVmerge$reg.mat.y)

Group1up <- subset(Group1Expr, (Group1Expr$C9 > 0.5 &
                                  Group1Expr$sALS > 0.5 &
                                  Group1Expr$FTLD > 0.5 &
                                  Group1Expr$VCP > 0.5 &
                                  Group1Expr$PET > 0.5 &
                                  Group1Expr$RAV > 0.5))

Group1down <- subset(Group1Expr, (Group1Expr$C9 < -0.5 &
                                    Group1Expr$sALS < -0.5 &
                                    Group1Expr$FTLD < -0.5 &
                                    Group1Expr$VCP < -0.5 &
                                    Group1Expr$PET < -0.5 &
                                    Group1Expr$RAV < -0.5))

Group1samedir <- rbind(Group1up, Group1down)
Group1samedir$corMean <- rowMeans(Group1samedir, na.rm = FALSE, dims = 1)
write.csv(Group1samedir, "CORRECTEDG1_samedir_mean.csv", quote = F)
```

This correction left a module of 72 genes with 95 edges

```{r, out.width = "900px", out.height = "700px", echo=F}
knitr::include_graphics("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/RMETHOD/RMETHOD.pdf")
```

## Running the method in R

Now that I had fixed the issue in the Cytoscape method, I needed to generate the same experiment but purely run in R code. This was relatively straightforward, I had to read in the correlation information with the combined "Gene1:Gene2" column (hence "combo"), and then find which rows within this column were present in all 6 datasets (Commonedge). I had double checked that the Gene ordering was correct in each dataset, as intersect requires an exact string match. 

I then subsetted the original table by the common rows, leaving 6 tables with identical rows. I ordered them alphabetically so that when I combined them, they would be identical. I did try to use merge to do this but it turned out to be way more complicated and messed with the formatting of the columns when they came out the other end. 

Once this Table was created (rownames are the gene relationships, columns are each of the Rho values corresponding to that relationship in each dataset) I could index out the rows that were either all positively correlated or all negatively correlated. This left 120 positively correlated gene-gene relationships, and 16 negatively correlated relationships.

```{r, eval=F}
setwd("~/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/")
C9_cor.5 <- read.csv("C9_cor.5_cytoscape_combo.csv")
sALS_cor.5 <-read.csv("sALS_cor.5_cytoscape_combo.csv")
FTLD_cor.5 <-read.csv("FTLD_cor.5_cytoscape_combo.csv")
VCP_cor.5 <-read.csv("VCP_cor.5_cytoscape_combo.csv")
PET_cor.5 <-read.csv("PET_cor.5_cytoscape_combo.csv")
RAV_cor.5 <-read.csv("RAV_cor.5_cytoscape_combo.csv")

Commonedge <- Reduce(intersect, list(C9_cor.5$combo, sALS_cor.5$combo, FTLD_cor.5$combo,
                                     VCP_cor.5$combo, PET_cor.5$combo, RAV_cor.5$combo))


#Subset each dataset with these common names so they are all the same size
C9_CE <- subset(C9_cor.5, C9_cor.5$combo %in% Commonedge)
C9_CE <- C9_CE[order(C9_CE$combo),]
sALS_CE <- subset(sALS_cor.5, sALS_cor.5$combo %in% Commonedge)
sALS_CE <- sALS_CE[order(sALS_CE$combo),]
FTLD_CE <- subset(FTLD_cor.5, FTLD_cor.5$combo %in% Commonedge)
FTLD_CE <- FTLD_CE[order(FTLD_CE$combo),]
VCP_CE <- subset(VCP_cor.5, VCP_cor.5$combo %in% Commonedge)
VCP_CE <- VCP_CE[order(VCP_CE$combo),]
PET_CE <- subset(PET_cor.5, PET_cor.5$combo %in% Commonedge)
PET_CE <- PET_CE[order(PET_CE$combo),]
RAV_CE <- subset(RAV_cor.5, RAV_cor.5$combo %in% Commonedge)
RAV_CE <- RAV_CE[order(RAV_CE$combo),]



CommonGroup <- data.frame(row.names = C9_CE$combo,
                         C9 = C9_CE$reg.mat,
                         sALS = sALS_CE$reg.mat,
                         FTLD = FTLD_CE$reg.mat,
                         VCP = VCP_CE$reg.mat,
                         PET = PET_CE$reg.mat,
                         RAV = RAV_CE$reg.mat)


CG_conserved_up <- CommonGroup[apply(CommonGroup, MARGIN = 1, function(x) all(x > 0)), ]
CG_conserved_down <- CommonGroup[apply(CommonGroup, MARGIN = 1, function(x) all(x < 0)), ]

CG_samedir <- rbind(CG_conserved_up, CG_conserved_down)
CG_samedir$corMean <- rowMeans(CG_samedir, na.rm = FALSE, dims = 1)
write.csv(CG_samedir, "RMETHOD_samedir_mean.csv", quote = F)
```

At first I was confused that there were more relationships than were reported using the cytoscape method, until I realised that there was the further step of taking just the larger module of the 72 genes and 95 relationships within that result. The networks are identical, which is good support for the method.

```{r, out.width = "900px", out.height = "700px", echo=F}
knitr::include_graphics("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/RMETHOD/RMETHOD_wholegroup.csv.pdf")
```
## Control calculations
The control results are the same as above

##Relationship with known disease genes
As the network is now slightly smaller, I had to re-run the previous analyses. First of all, we no longer have ELP3 as one of the diseased genes in the module. Secondly I re-ran the analysis for PPI interaction between the nugget and known disease genes that can cause TDP-43 pathology.

```{r, out.width = "900px", out.height = "700px", echo=F}
knitr::include_graphics("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/RMETHOD/disgene_cor_PPI.pdf")
```

Blue genes: Common Coexpression Module   
Yellow genes: Disease genes   
Pink genes: Both in Coexpression Module and Disease Gene   
Edges: PPI relationships   
Red Edges: Relationships between coexpression genes and disease genes   

## Merging with coexpression module network

This helps to see how the PPI module relates to the coexpression module.
```{r, out.width = "900px", out.height = "700px", echo=F}
knitr::include_graphics("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/RMETHOD/Union_Nugget+PPI.pdf")
```

Blue genes: Common Coexpression Module   
Yellow genes: Disease genes   
Pink genes: Both in Coexpression Module and Disease Gene   
Coloured Edges: Coexpression (red positive, blue negative)   
Black Edges: Protein-Protein interaction   

## Enrichment

Enrichment of certain gene lists will help support the role of this module as important in TDP-43 pathology and neurodegeneration in general. Enrichment is calculated as follows:

1) Select a random set of genes of the same length as the PPI network (2550)
2) Of that set, select a random set of genes the same length as the module (72)
3) Find the intersect of the random module and the list of interest
4) Store overlap
5) Repeat 100,000 times
6) Calculate number of times random set has higher overlap than your experiment and divide by 100,000 to get p value

```{r, eval=F}
###Random permutations for Differential Expression##
options(scipen=999)

DEGPPI <- readLines("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/DEG_PPI_Genes_nofib.txt")
DEGs <- readLines("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/FoldChangeResults/Filtered_upanddown.txt")
ALSOD <- readLines("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/ALSoDgenes.txt")
Disgenes <- readLines("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/TDP-43genes.txt")
Taylor <- readLines("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/M&R/Taylor_TDP43.txt")
nugget <- readLines("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/RMETHOD/NuggetGenes.txt")

#Load file with all genes
library(hgu133plus2.db)
sym <- hgu133plus2SYMBOL
sym1 <- mappedkeys(sym)
sym2 <- as.list (sym[c(sym1)]) 
sym3 <- data.frame (sym2)
sym.probes <- names (sym2)
sym.genes <- sym3[1,]

sym.genes <- t(sym.genes)

allgenes <- sym.genes[!duplicated(sym.genes),]


setwd("~/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/")
C9_cor.5 <- read.csv("C9_cor.5_cytoscape.csv")
sALS_cor.5 <-read.csv("sALS_cor.5_cytoscape.csv")
FTLD_cor.5 <-read.csv("FTLD_cor.5_cytoscape.csv")
VCP_cor.5 <-read.csv("VCP_cor.5_cytoscape.csv")
PET_cor.5 <-read.csv("PET_cor.5_cytoscape.csv")
RAV_cor.5 <-read.csv("RAV_cor.5_cytoscape.csv")

#Find length of overlap
length(intersect(nugget, Taylor))

#indicate the number of overlapping genes identified by DE analysis
test <- 4
samplenum <- 72
samplelist <- Taylor

m=100000 #number of repetitions 
r <- c(1:m) #store repetition numbers in vector "r"


for (j in 1:m){
  PPIsample <- sample(allgenes, size=2550, replace=FALSE)
  modulesample <- sample(PPIsample, size = samplenum, replace = FALSE)
  random <- Reduce(intersect, list(modulesample, samplelist))
  r[j] <- length(random)
}

test1 <- which(r > test)  # count number of times r is larger than test value
result <- sum((length(test1)+1))/(m+1) # calculate P value
result
mean(r)
range(r)
```

###TDP-43 PPI genes (Taylor)
Number of genes in module = 4

p = 0.001169988   
mean of random overlap = 0.78322   
range of random overlap = 0-7   

```{r, out.width = "900px", out.height = "700px", echo=F}
knitr::include_graphics("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/RMETHOD/TaylorOverlap.png")
```

### Disease genes
Number of genes in module = 2

p = 0.009109909   
mean of random overlap = 0.42512   
range of random overlap = 0-6   

```{r, out.width = "900px", out.height = "700px", echo=F}
knitr::include_graphics("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/PPI_Network/Coexpression/PPI_Coexpression/CorrelationValue/RMETHOD/DisgeneOverlap.png")
```
