---
title: "Untitled"
author: "Claire Green"
output: pdf_document
---

```{r, echo=FALSE}
library(knitr)
opts_chunk$set(tidy.opts=list(width.cutoff=60))
```

#Monday

On Monday I gave my presentation to SITraN at the monday afternoon seminar. It went down pretty well, most comments were to John on his biomarker model. I was asked about interactions with TDP-43 but apparently the list we have here is pretty large so may screw with enrichment calculations. Depends on whether everything interacts with TDP-43!

This afternoon I had a look at EvoTol, however I'm not entirely sure what the output means. There isn't great documentation and the paper doesn't really help. I know it gives a score but I'm not sure if a high score or a low score is what I'm looking for...

In my meeting with Win we had some deep conversations about how to take this further. I need to look at how to generate a Q plot to show the difference between enrichment of a random set and my set at different thresholds. I remember generating Q plots when I was learning R, I'm just not sure how to automate the process. 

Win also said that I need to look at the regulatory side of the model. I was a little lost in what he was saying but he recommended a paper by Goldstein and Californo (2015)

#Tuesday
Today I looked at generating the QQ plot that Win wanted me to create. To do so I need to show the p values of enrichment at each threshold as compared to what you would expect by chance. The problem I faced was that to generate my list, I used both HGNC and Ensembl IDs to generate the list, and now that I am repeating it I cannot seem to get the same results again. The first time I tried using the files from "TopGenes_ensemblID" for the Microarray and "16_3_21" for the RNA seq. When I use the pre-created files from these files for the consensus, I get the same results as I got before. However when I use the ranked result .csv file for the microarray data, I don't. I tried conducting the analysis from scratch for the microarray data (see file EnsemblID_0516), but somehow I end up with a huge amount more Ensembl IDs than I did before.

This is my code 

```{r, eval=FALSE}
##Differential Expression of Genes##

setwd("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Data/GeneExpressionAnalysis/Microarray/FTLD/")
library(widgetTools)
library(tcltk)
library(DynDoc)
library(tools)
library(affy)
library(Biobase)
library(tkWidgets)

#run program to choose .CEL files from directory
celfiles <- fileBrowser(textToShow = "Choose CEL files", testFun = hasSuffix("[cC][eE][lL]"))
#celfiles<-basename(celfiles)
Data<-ReadAffy(filenames=celfiles) #read in files
rmaEset<-rma(Data) #normalise using RMA
analysis.name<-"FTLD" #Label analysis
dataMatrixAll<-exprs(rmaEset) #takes expression from normalised expression set


#mas5call generates presence/absence calls for each probeset
mas5call<-mas5calls(Data)
callMatrixAll<-exprs(mas5call)
colnames(callMatrixAll)<-sub(".CEL", ".mas5-Detection", colnames(callMatrixAll),fixed=TRUE)
colnames(callMatrixAll)<-sub(".cel", ".mas5-Detection", colnames(callMatrixAll),fixed=TRUE)
callMatrixAll<-as.data.frame(callMatrixAll)
callMatrixAll$ProbeSetID<-rownames(callMatrixAll)
countPf<-function(x){
  sum(x=="P")
}

#count how many samples have presence calls
countPl<-apply(callMatrixAll, 1, countPf)
callMatrixAll$ProbeSetID<-rownames(callMatrixAll)
countPdf<-data.frame(ProbeSetID=names(countPl), countP=countPl) 

#read annotation file

###USING BIOMART
# library (biomaRt)
# mart <- useMart("ENSEMBL_MART_ENSEMBL",dataset="hsapiens_gene_ensembl", host="www.ensembl.org") 
# x <- rownames(dataMatrixAll) #create vector containing probe IDs
# mart_attribute <- listAttributes(mart)
# annotation <- getBM(attributes=c("affy_hg_u133a_2", "hgnc_symbol", "description"), 
#                    filters = "affy_hg_u133a_2", values = x, mart = mart)
# annotation<-subset(annotation, subset=(hgnc_symbol !="")) #if no gene symbol, discount

# USING ANNOTATION FILE (if .csv, convert to .txt using excel)
#annotation.file<-"/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Data/HG-U133_Plus_2.na35.annot.csv/HG-U133_Plus_2.na35.annot.txt"
annotation.file<-"/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Data/HG-U133A_2.na35.annot.csv/HG-U133A_2.na35.annot.txt"
annotation<-read.table(annotation.file, header = TRUE, row.names=NULL, sep="\t", skip=0, stringsAsFactors=F, quote = "", comment.char="!", fill = TRUE, as.is = TRUE)
dim(annotation)
nrow(annotation)
#[1] 39699
annotation<-subset( annotation, subset=(Ensembl !="---")) #if no gene symbol, discount

# Remove rows in which genes are noted to have negative strand matching probes
idxNegativeStrand<-grep("Negative Strand Matching Probes", annotation$Annotation.Notes)
if(length(idxNegativeStrand)>0)
{
  annotation<-annotation[-idxNegativeStrand,]
}


expressionMatrix<-exprs(rmaEset)
colnames(expressionMatrix)

#this is for matched samples
#tonsil<-factor(c("T101","T101","T102","T102","T103","T103"))
Treat<-factor(rep(c("Control", "Patient"),c(8,16)), levels=c("Control", "Patient"))
design<-model.matrix(~Treat)
rownames(design)<-colnames(expressionMatrix)
design

#Conduct statistical analysis of expression
library(limma)
fit<-lmFit(expressionMatrix, design) #linear model fit
fit<-eBayes(fit) 
result<-topTable(fit, coef="TreatPatient", adjust="BH", number=nrow(expressionMatrix)) #"BH" adjust for multiple hypothesis testing
#toptable normally takes top number but this takes all


result$"ProbeSetID"<-rownames(result) #make probeset IDs the row names
head(result$"ProbeSetID") 
result$"Fold Change"<-2^result$logFC 
result$"Fold Change"[result$"Fold Change"<1]<-(-1)/result$"Fold Change"[result$"Fold Change"<1] #converts log fold change into a linear value above or below 0
expressionLinear<-as.data.frame(2^expressionMatrix)
expressionLinear$ProbeSetID<-rownames(expressionLinear)
result<-merge(result, expressionLinear, by.x="ProbeSetID", by.y="ProbeSetID") #merge values into one array
result<-merge(annotation, result, by.x="Probe.Set.ID", by.y="ProbeSetID")
result<-merge(result, countPdf, by.x="Probe.Set.ID", by.y="ProbeSetID")

setwd(dir = "/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Data/GeneExpressionAnalysis/Microarray/EnsemblID_0516")
write.csv(result, file=paste(analysis.name, "result.csv", sep=""), sep="\t", row.names=FALSE, quote = FALSE)

result<-subset(result, Gene.Symbol!="") #removes any probes for which there are no gene symbols
result<-subset(result, subset=(countP>2)) #only takes results that have at least 2 samples with a presence call for a probe



# nrow(result)
# foldchange<-1.5
# pvalue<-0.05
# #adj_P_Val<-0.05
# siggenes<-subset(result, subset=(P.Value < pvalue) & abs(logFC) > log2(foldchange))
# #siggenes<-subset(result, subset=(adj.P.Val < 0.05))
# nrow(siggenes)
# siggenesup<-subset(siggenes, subset= logFC > 0)
# siggenesdown<-subset(siggenes, subset=logFC < 0)
# colnames(siggenesup)
# nrow(siggenesup)
# nrow(siggenesdown)
# UpandDown<-intersect(siggenesup$"Gene.Symbol", siggenesdown$"Gene.Symbol")
# length(UpandDown)
# 
# UporDown<-subset(siggenes, subset=(!siggenes$"Gene.Symbol"%in% UpandDown))
# upsiggenes<-subset(siggenesup, subset=(!siggenesup$"Gene.Symbol"%in% UpandDown))
# downsiggenes<-subset(siggenesdown, subset=(!siggenesdown$"Gene.Symbol"%in% UpandDown))
# length(unique(siggenes$"Gene.Symbol"))
# uniquesiggenes <- unique(siggenes$Gene.Symbol)
# length(unique(upsiggenes$"Gene.Symbol"))
# length(unique(downsiggenes$"Gene.Symbol"))
# length(unique(UporDown$"Gene.Symbol"))


###Write results to CSV files for consensus analysis
setwd("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Data/GeneExpressionAnalysis/Microarray/EnsemblID_0516/")
#dir.create(paste("TopGenes", Sys.Date(), sep = "_")) #create directory using the day's date
#Take results, remove duplicate rows for genes, order by adjusted p value and take top X number of genes
uniqueresult <- result[!duplicated(result[,18]),]

#For ordering by adjusted p value

genesort <- uniqueresult[order(uniqueresult$adj.P.Val),]
write.csv(genesort, file=paste(analysis.name, "rankeduniqueresult.csv", sep=""), sep="\t", row.names=FALSE, quote = FALSE)
 

topgene <- genesort[1:500,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_500.csv"))
topgene <- genesort[1:1000,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_1000.csv"))
topgene <- genesort[1:1500,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_1500.csv"))
topgene <- genesort[1:2000,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_2000.csv"))
topgene <- genesort[1:2500,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_2500.csv"))
topgene <- genesort[1:3000,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_3000.csv"))
topgene <- genesort[1:3500,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_3500.csv"))
topgene <- genesort[1:4000,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_4000.csv"))
topgene <- genesort[1:4500,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_4500.csv"))
topgene <- genesort[1:5000,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_5000.csv"))
topgene <- genesort[1:5500,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_5500.csv"))
topgene <- genesort[1:6000,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_6000.csv"))
topgene <- genesort[1:6500,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_6500.csv"))
topgene <- genesort[1:7000,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_7000.csv"))
topgene <- genesort[1:7500,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_7500.csv"))
topgene <- genesort[1:8000,]
write.csv(x = topgene, file = paste(analysis.name,"_ap_8000.csv"))




# #For ordering by fold change
# genesort <- uniqueresult[order(uniqueresult$`Fold Change`),]
# topgene <- genesort[1:500,]
# 
# genesort <- uniqueresult[order(-uniqueresult$`Fold Change`),]
# botgene <- genesort[1:500,]
# 
# topFC <- rbind(topgene,botgene)
# 
# write.csv(x = topFC, file = "VCP_fc_1000")
# 
# genesort <- uniqueresult[order(uniqueresult$`Fold Change`),]
# topgene <- genesort[1:1000,]
# 
# genesort <- uniqueresult[order(-uniqueresult$`Fold Change`),]
# botgene <- genesort[1:1000,]
# 
# topFC <- rbind(topgene,botgene)
# 
# write.csv(x = topFC, file = "VCP_fc_2000")
# 
# genesort <- uniqueresult[order(uniqueresult$`Fold Change`),]
# topgene <- genesort[1:1500,]
# 
# genesort <- uniqueresult[order(-uniqueresult$`Fold Change`),]
# botgene <- genesort[1:1500,]
# 
# topFC <- rbind(topgene,botgene)
# 
# write.csv(x = topFC, file = "VCP_fc_3000")
# 
# genesort <- uniqueresult[order(uniqueresult$`Fold Change`),]
# topgene <- genesort[1:2000,]
# 
# genesort <- uniqueresult[order(-uniqueresult$`Fold Change`),]
# botgene <- genesort[1:2000,]
# 
# topFC <- rbind(topgene,botgene)
# 
# write.csv(x = topFC, file = "VCP_fc_4000")






# dir.create(paste("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43 Data Sets/DE Genes/CHMP2B/unique"))
# setwd("/Users/clairegreen/Documents/PhD/TDP-43/TDP-43 Data Sets/DE Genes/CHMP2B/unique")
# 
# write.table(siggenes, file=paste(analysis.name," RMA limma siggenes biomart p_", pvalue, "_fold change_", foldchange, ".txt", sep=""), sep="\t", row.names=FALSE, quote = FALSE)
# write.table(upsiggenes, file=paste(analysis.name," RMA limma upsiggenes biomart p_", pvalue, "_fold change_", foldchange, ".txt", sep=""), sep="\t", row.names=FALSE, quote = FALSE)
# write.table(downsiggenes, file=paste(analysis.name," RMA limma downsiggenes biomart p_", pvalue, "_fold change_", foldchange, ".txt", sep=""), sep="\t", row.names=FALSE, quote = FALSE)
# write.table(uniquesiggenes, file=paste(analysis.name,"unique RMA limma siggenes biomart p_", pvalue, "_fold change_", foldchange, ".txt", sep=""), sep="\t", row.names=FALSE, quote = FALSE)

```

I managed to recreate the lists (but somehow discovered 6 new genes at a threshold of 5000...). It took all day.

1) Files were generated using the Wenbin_DE_Gene code. Criteria have been defined in previous lab books. Genes must have both hgnc and ENS symbol and not be duplicated. 

The files for microarray gene symbols are in /Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Data/GeneExpressionAnalysis/Microarray/TopGenes_2016-02-15. The files for microarray ensembl IDs are in /Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Data/GeneExpressionAnalysis/Microarray/TopGenes_ensemblID. Rows without ENS symbols needed to be removed. Files for RNA seq gene symbols are in /Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Data/GeneExpressionAnalysis/RNA-seq/16_3_17, files for RNA seq ENS symbols are in /Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Data/GeneExpressionAnalysis/RNA-seq/16_3_21. 

List of all genes is in file "QQresulttest3" in /Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/QQ/Test3.

Here is the code I used to take the different thresholds

```{r, eval=FALSE}
#Create list for C9orf72#
genelist <- C9DEgene$Ensembl
thresh <- c(500, 1000, 1500, 2000, 2500, 3000, 3500, 4000, 4500, 5000, 5500, 6000, 6500, 7000, 7500, 8000)
resultC9 <- list(a = 1:500, b = 1:1000, c = 1:1500, d = 1:2000, e = 1:2500, f = 1:3000, g = 1:3500, 
               h = 1:4000, i = 1:4500, j = 1:5000, k = 1:5500, l = 1:6000, m = 1:6500, n = 1:7000, 
               o = 1:7500, p = 1:8000)

for (i in 1:length(thresh)) {
  resultC9[[i]] <- genelist[1:thresh[[i]]]
}

#Create list for CHMP2B#
genelist <- CHDEgene$Ensembl
thresh <- c(500, 1000, 1500, 2000, 2500, 3000, 3500, 4000, 4500, 5000, 5500, 6000, 6500, 7000, 7500, 8000)
resultCH <- list(a = 1:500, b = 1:1000, c = 1:1500, d = 1:2000, e = 1:2500, f = 1:3000, g = 1:3500, 
                h = 1:4000, i = 1:4500, j = 1:5000, k = 1:5500, l = 1:6000, m = 1:6500, n = 1:7000, 
                o = 1:7500, p = 1:8000)

for (i in 1:length(thresh)) {
  resultCH[[i]] <- genelist[1:thresh[[i]]]
}

#Create list for sALS#
genelist <- sALSDEgene$Ensembl
thresh <- c(500, 1000, 1500, 2000, 2500, 3000, 3500, 4000, 4500, 5000, 5500, 6000, 6500, 7000, 7500, 8000)
resultsALS <- list(a = 1:500, b = 1:1000, c = 1:1500, d = 1:2000, e = 1:2500, f = 1:3000, g = 1:3500, 
                h = 1:4000, i = 1:4500, j = 1:5000, k = 1:5500, l = 1:6000, m = 1:6500, n = 1:7000, 
                o = 1:7500, p = 1:8000)

for (i in 1:length(thresh)) {
  resultsALS[[i]] <- genelist[1:thresh[[i]]]
}

#Create list for FTLD#
genelist <- FTLDDEgene$Ensembl
thresh <- c(500, 1000, 1500, 2000, 2500, 3000, 3500, 4000, 4500, 5000, 5500, 6000, 6500, 7000, 7500, 8000)
resultFTLD <- list(a = 1:500, b = 1:1000, c = 1:1500, d = 1:2000, e = 1:2500, f = 1:3000, g = 1:3500, 
                h = 1:4000, i = 1:4500, j = 1:5000, k = 1:5500, l = 1:6000, m = 1:6500, n = 1:7000, 
                o = 1:7500, p = 1:8000)

for (i in 1:length(thresh)) {
  resultFTLD[[i]] <- genelist[1:thresh[[i]]]
}

#Create list for VCP#
genelist <- VCPDEgene$Ensembl
thresh <- c(500, 1000, 1500, 2000, 2500, 3000, 3500, 4000, 4500, 5000, 5500, 6000, 6500, 7000, 7500, 8000)
resultVCP <- list(a = 1:500, b = 1:1000, c = 1:1500, d = 1:2000, e = 1:2500, f = 1:3000, g = 1:3500, 
                h = 1:4000, i = 1:4500, j = 1:5000, k = 1:5500, l = 1:6000, m = 1:6500, n = 1:7000, 
                o = 1:7500, p = 1:8000)

for (i in 1:length(thresh)) {
  resultVCP[[i]] <- genelist[1:thresh[[i]]]
}

#Create list for Petrucelli#
genelist <- PETDEgene$
thresh <- c(500, 1000, 1500, 2000, 2500, 3000, 3500, 4000, 4500, 5000, 5500, 6000, 6500, 7000, 7500, 8000)
resultPET <- list(a = 1:500, b = 1:1000, c = 1:1500, d = 1:2000, e = 1:2500, f = 1:3000, g = 1:3500, 
                h = 1:4000, i = 1:4500, j = 1:5000, k = 1:5500, l = 1:6000, m = 1:6500, n = 1:7000, 
                o = 1:7500, p = 1:8000)

for (i in 1:length(thresh)) {
  resultPET[[i]] <- genelist[1:thresh[[i]]]
}

#Create list for Ravits#
genelist <- RAVDEgene$Row.names
thresh <- c(500, 1000, 1500, 2000, 2500, 3000, 3500, 4000, 4500, 5000, 5500, 6000, 6500, 7000, 7500, 8000)
resultRAV <- list(a = 1:500, b = 1:1000, c = 1:1500, d = 1:2000, e = 1:2500, f = 1:3000, g = 1:3500, 
                h = 1:4000, i = 1:4500, j = 1:5000, k = 1:5500, l = 1:6000, m = 1:6500, n = 1:7000, 
                o = 1:7500, p = 1:8000)

for (i in 1:length(thresh)) {
  resultRAV[[i]] <- genelist[1:thresh[[i]]]
}

##Intersect lists of same length##
result <- list(list(a = 1:5000, b = 1:5000, c = 1:5000, d = 1:5000, e = 1:5000, f = 1:5000, g = 1:5000, 
                    h = 1:5000, i = 1:5000, j = 1:5000, k = 1:5000, l = 1:5000, m = 1:5000, n = 1:5000, 
                    o = 1:5000, p = 1:5000))

for (i in 1:16) {
  result[[i]] <- Reduce(intersect, list(resultC9[[i]],resultCH[[i]],resultsALS[[i]],resultFTLD[[i]],
                                        resultVCP[[i]],resultPET[[i]],resultRAV[[i]]))
}

##Make list values equal, convert to data frame and save##
for (i in 1:16) {
  length(result[[i]]) <- 1251
}

setwd(dir = "/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/QQ/Test3/")
resultdf <- as.data.frame(result)
write.csv(result, file = "QQresultENS3.csv")

#save individual lists as data frames so they can be converted in biomart.R
result2000 <- as.data.frame(result[[4]])
result2500 <- as.data.frame(result[[5]])
result3000 <- as.data.frame(result[[6]])
result3500 <- as.data.frame(result[[7]])
result4000 <- as.data.frame(result[[8]])
result4500 <- as.data.frame(result[[9]])
result5000 <- as.data.frame(result[[10]])
result5500 <- as.data.frame(result[[11]])
result6000 <- as.data.frame(result[[12]])
result6500 <- as.data.frame(result[[13]])
result7000 <- as.data.frame(result[[14]])
result7500 <- as.data.frame(result[[15]])
# result8000 <- as.data.frame(result[[16]])

Z <- read.csv(file = "QQresulttest3.csv", na.strings = c("", "NA)"))
Z <- as.list(Z)
Z<- lapply(Z, function(x) x[!is.na(x)])
Z <- lapply(Z, function(x) x[!duplicated(x)])
```

To convert Ensembl IDs to HGNC symbols I used this code:

```{r, eval=FALSE}

library (biomaRt)
mart <- useMart("ENSEMBL_MART_ENSEMBL",dataset="hsapiens_gene_ensembl", host="www.ensembl.org")



analysis.name <- "7500"
genes <- overlap 
mart_attribute <- listAttributes(mart)
mart_filter <- listFilters(mart)
annotation <- getBM(attributes=c("ensembl_gene_id", "hgnc_symbol"),
                   filters = "ensembl_gene_id", values = genes, mart = mart)

setwd(dir = "/Users/clairegreen/Documents/PhD/TDP-43/TDP-43_Code/Results/GeneExpression/QQ/Test3/GeneIds/")
write.csv(annotation, paste(analysis.name,"geneIDs.csv"))
```

#Wednesday

With my list, I looked for enrichment at each level. I used FishersExact_Enrichment.R. Results can be found in the file "Enrichment results". With Fisher's exact test, enrichment is only seen in ExAC and the Pasterkamp and Taylor PPI. 

I conducted the same enrichment test with the hyperpathway function but it gives much better p values which I'm a little sceptical of. Maybe I need to reanalyse the pathways, I would imagine it would have to be consistent. 


